sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client_options.hpp

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Defines the supported options to create a Attestation clients.
+ *
+ */
+
+#pragma once
+
+#include "azure/core/internal/client_options.hpp"
+
+#include "attestation_client_models.hpp"
+#include "dll_import_export.hpp"
+
+namespace Azure { namespace Security { namespace Attestation {
+
+  class ServiceVersion final {
+  private:
+    std::string m_version;
+
+  public:
+    /**
+     * @brief Construct a new Service Version object
+     *
+     * @param version The string version for the Attestation service.
+     */
+    ServiceVersion(std::string version) : m_version(std::move(version)) {}
+
+    /**
+     * @brief Enable comparing the ext enum.
+     *
+     * @param other Another #ServiceVersion to be compared.
+     */
+    bool operator==(ServiceVersion const& other) const { return m_version == other.m_version; }
+
+    /**
+     * @brief Return the #ServiceVersion string representation.
+     *
+     */
+    std::string const& ToString() const { return m_version; }
+
+    /**
+     * @brief Use to send request to the 2020-10-01 version of Attestation service.
+     *
+     */
+    AZ_ATTESTATION_DLLEXPORT static const ServiceVersion V2020_10_01;
+  };
+
+  /// The TokenValidationCallbackFn represents a callback which is called to allow the caller
+  /// to perform additional token validation options beyond the validations performed by the
+  /// attestation SDK.
+  using TokenValidationCallbackFn = std::function<
+      void(std::string const& rawToken, Models::AttestationSigner const& tokenSigner)>;
+
+  /** @brief The AttestationTokenValidationOptions represents a set of options which control how
+   * attestation tokens are validated. */
+  struct AttestationTokenValidationOptions final
+  {
+    /// Controls whether attestation tokens are validated at all.
+    ///
```

Comment:
/**
     * @brief Define policies to be called one time for every HTTP request from an SDK client.
     *
     */